tests/_singerlib/test_catalog.py

```diff
@@ -0,0 +1,267 @@
+from typing import List, Optional
```

Comment:
Added some more tests here for catalog and metadata manipulation to ensure 100% coverage in the `singer_sdk._singerlib` module.